Analysis/src/EventTags.cpp

```diff
@@ -5,36 +5,79 @@ This file is part of https://github.com/hh-italian-group/hh-bbtautau. */
 #include "h-tautau/Cuts/include/hh_bbtautau_Run2.h"
 #include <boost/property_tree/ptree.hpp>
 #include <boost/property_tree/json_parser.hpp>
+/*
 
+1. in the constructor you load json into std::map<std::pair<UncSource, UncScale>, float> ---> Fatto
+2. when you calculate btag weight you know what is uncSource and uncScale -> multiply it by the value from the map
+for btag related unc, you should multiply by both sf and unc up/down correction
+*/
 
 namespace analysis {
 namespace bbtautau {
 
+std::string get_name(UncertaintySource unc_source, UncertaintyScale unc_scale){
+    std::string complete_name;
+    if(unc_scale!= UncertaintyScale::Central && unc_scale!= UncertaintyScale::Up && unc_scale!= UncertaintyScale::Down)
+        throw exception("Incorrect uncertainty scale")
+
+    if(unc_source==UncertaintySource::None){
+        if(unc_scale==UncertaintyScale::Central)
+            complete_name="Central";
+        else
+            throw exception("Central Value can't have different scales");
+    }
+    else{
+        if(unc_scale!=UncertaintyScale::Central)
+            complete_name=EnumToString(unc_source)+"_"+EnumToString(unc_scale);
+        else{
+            complete_name=EnumToString(unc_source);
+        }
+    }
+    return complete_name;
+
+}
+
 EventTagCreator::EventTagCreator(const EventCategorySet& _categories, const EventSubCategorySet& _subCategories,
                                   const std::set<UncertaintySource>& _event_unc_sources,
                                   const std::set<UncertaintySource>& _norm_unc_sources,
-                                  bool _use_IterativeFit, std::string _file_name) :
+                                  bool _use_IterativeFit, std::string _json_file)  :
         categories(_categories), subCategories(_subCategories), event_unc_sources(_event_unc_sources),
-        norm_unc_sources(_norm_unc_sources), use_IterativeFit(_use_IterativeFit), file_name(_file_name)
+        norm_unc_sources(_norm_unc_sources), use_IterativeFit(_use_IterativeFit),
+        json_file(_json_file),
 {
 
-    std::size_t in = file_name.find("/201"); // --> in order to remove all the path before the file name
-    file_name.erase(0,in+1);
-    std::size_t fin = file_name.find(".root"); // --> to remove the last part of the file
-    file_name.erase(fin);
-    std::vector<std::string> file= SplitValueList(file_name, false, "_", true);
+
     if(use_IterativeFit) {
         boost::property_tree::ptree loadPtreeRoot;
-        std::string path_to_json_file = "r_factors.json";
-        boost::property_tree::read_json(path_to_json_file, loadPtreeRoot);
-        boost::property_tree::ptree temp ;
-        if(loadPtreeRoot.get_child(file_name).template get_value<float>()==0)
+        boost::property_tree::read_json(json_file, loadPtreeRoot);
+
+        std::map<std::pair<UncertaintySource, UncertaintyScale>, float> iterativeFit_corrections{
```

Comment:
I found do it another way around: read all unc available in the json in the map, don't list them here.